### DISCUSSION TITLE
Missing or incorrect documentation for field/packet type

### DISCUSSION THREAD

**Discussion by ghost** [Pryaxis/TShock#2640] [Q&A]
# Incorrect behaviour

The problem is `Connect request [1]` - https://tshock.readme.io/docs/multiplayer-packet-structure#connect-request-1
If client send payload string without leading byte length, then server will disconnect it with the LegacyMultiplayer reason.

# Official terraria client behaviour

Official terraria client sends version with the leading byte length

# Payload Example (Packet headers omitted)

`0xB 0x54 0x65 0x72 0x72 0x61 0x72 0x69 0x61 0x32 0x34 0x38` (b'\x0bTerraria248') - valid
`0x54 0x65 0x72 0x72 0x61 0x72 0x69 0x61 0x32 0x34 0x38` (b'Terraria248') - invalid (LegacyMultiplayer.4)

# Suggestion

If String is encoded like: `length + utf8_bytes` please note that. Is documentation contains structs without details like missing/incorrect field/size/offset?

**Comment by punchready** [comment]
First of all, a bunch of packets contain strings, not only the first one.
Second of all, the (presumed) reason why this is not documented is that .NET `BinaryReader/Writer.Read/Write(string)` will transparently read/write the string length in 7 bit encoding, meaning the developer will probably never even come in contact with that.

**Reply by bartico6** [reply]
This has already been raised in the related issue (#2641)

**Reply by punchready** [reply]
Ah sorry i didn't see that - my bad
